### ISSUE TITLE
Add pre-commit hooks to repository

### ISSUE THREAD

**Issue by justinmayer** [getpelican/pelican#1805]
In other projects I've used the open-source [Pre-commit](http://pre-commit.com) package to enforce passing tests and `flake8` compliance. Since the latter will soon be in place thanks to @ingwinlu (#1753), I wonder whether adding pre-commit hooks to the Pelican repository might be sensible. There are already a significant number of pull requests that are submitted with test failures, and asking folks to comply with PEP-8 will likely compound the number of "violations" that land in the PR queue. If these checks are automatically performed before a commit can be made, however, PR quality should increase significantly.

What does everyone think? (cc: @ametaireau @avaris @iKevinY @kylef)

**Comment by ingwinlu** [comment]
we provide a lot of documentation and tooling to produce pr's that are valuable. If someone comes along and just slings something at the git repository it is annoying, but it might be valuable in pointing out inconsistencies with documentation or unclear behavior (or even missing features).

If you require passing tests all this little indications on what might be wrong can no longer pop up, since people who just pass over it won't get to publish it.

to keep those often stale pr's and issues away I would rather close them when they 'timeout' than to prevent them in the first place (or don't follow the contribution guidelines for issues)

tl;dr: provide easy tooling (tox -e py27,34) good, enforcing them that hard - meh.

ps: there are a lot of 'passing' pr's open that are bad as well...

**Comment by leotrs** [comment]
Agree with @ingwinlu.

**Comment by justinmayer** [comment]
Good points. Thanks, folks.
